share/picongpu/unit/TWTSTight.cpp

```diff
@@ -0,0 +1,194 @@
+
+        forEach(
+            [&](uint32_t const idx)
+            {
+                auto valueIdx = blockIdx * T_numThreadsPerBlock + idx;
+                if(valueIdx < numValues)
+                {
+                    result(0u + valueIdx) = testEfield.calcTWTSFieldX(pos, time);
+                    result(1u + valueIdx) = testEfield.calcTWTSFieldY(pos, time);
+                    result(2u + valueIdx) = testEfield.calcTWTSFieldZ(pos, time);
+                    result(3u + valueIdx) = testBfield.calcTWTSFieldX(pos, time);
+                    result(4u + valueIdx) = testBfield.calcTWTSFieldY(pos, time);
+                    result(5u + valueIdx) = testBfield.calcTWTSFieldZ(pos, time);
+                }
+            });
+    }
+};
+
+/** Test TWTSTight laser functions
+ *
+ * Compares the on host and on device computed result to analytical results.
+ *
+ */
+struct twtsTightNumberTest
+{
+    void operator()()
+    {
+        using namespace ::pmacc;
+        const templates::twtstight::EField testEfield = templates::twtstight::EField(
+            0.0,
+            800.0e-9,
+            30.0e-15,
+            2.5e-6,
+            5. * (PI / 180.),
+            1.0,
+            0.0,
+            false,
+            30. * (PI / 180.));
+        const templates::twtstight::BField testBfield = templates::twtstight::BField(
+            0.0,
+            800.0e-9,
+            30.0e-15,
+            2.5e-6,
+            5. * (PI / 180.),
+            1.0,
+            0.0,
+            false,
+            30. * (PI / 180.));
+        using float_T = templates::twtstight::float_T;
+        using float3_T = ::pmacc::math::Vector<float_T, 3u>;
+
+        constexpr uint32_t numBlocks = 1;
+        constexpr uint32_t numThreadsPerBlock = 1;
+        constexpr uint32_t numThreads = numBlocks * numThreadsPerBlock;
+        constexpr uint32_t numValuesPerThread = 6;
+        constexpr uint32_t numValues = numThreads * numValuesPerThread;
+        const float3_64 pos = float3_64{1.0e-6, 1.0e-6, 100.0 * 1.5e-6};
+        const float_64 time = float_64(1.0e-15);
+
+        HostBuffer<float_T, 1u> resultHost(numValues);
+        DeviceBuffer<float_T, 1u> resultDevice(numValues);
+        resultDevice.setValue(float_T(0.0));
+
+        PMACC_LOCKSTEP_KERNEL(GenerateEvals<numThreadsPerBlock>{})
+            .template config<numThreadsPerBlock>(
+                numBlocks)(numValues, numValuesPerThread, pos, time, resultDevice.getDataBox());
+
+        resultHost.copyFrom(resultDevice);
+
+        auto res = resultHost.getDataBox();
+        auto hostEfield = float3_X(
+            testEfield.calcTWTSFieldX(pos, time),
+            testEfield.calcTWTSFieldY(pos, time),
+            testEfield.calcTWTSFieldZ(pos, time));
+        auto hostBfield = float3_X(
+            testBfield.calcTWTSFieldX(pos, time),
+            testBfield.calcTWTSFieldY(pos, time),
+            testBfield.calcTWTSFieldZ(pos, time));
+
```

Comment:
```suggestion

// This combination of compilers has a bug that is triggered by Catch2 internally suppressing warnings.
// See https://github.com/ComputationalRadiationPhysics/picongpu/pull/5174#issuecomment-2467890326
#if(__GNUC__ != 11 && (__CUDACC__ < 11000 || __CUDACC__ >= 12000))
```

Reply:
In prinziple we have BOOST macros for version control e.g.

```
#if BOOST_COMP_NVCC && BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(11, 3, 0)
```

but I am fine with this change from the PR.